lib/monitor.js

```diff
@@ -133,32 +143,56 @@ class Monitor {
         }
 
         // Events can not be any of ['log', 'ops', 'request', 'response', 'tail']
-        for (let i = 0; i < this.settings.extensions.length; ++i) {
-            const event = this.settings.extensions[i];
-            this._server.events.on(this.settings.extensions[i], (...args) => {
-
-                const payload = {
-                    event,
-                    timestamp: Date.now(),
-                    payload: args
-                };
-                this.push(() => Object.assign({}, payload));
+        this.settings.extensions.forEach((event) => {
+
+            const listener = (...args) => {
+
+                this.push(() => {
+
+                    return Object.assign({}, {
+                        event,
+                        timestamp: Date.now(),
+                        payload: args
+                    });
+                });
+            };
+
+            // Store a reference to the listener so we can remove them later
+            this._extensionListeners.push({
+                event, listener
             });
-        }
+
+            this._server.events.on(event, listener);
+        });
     }
 
     stop() {
 
         this._state.report = false;
-        if (this._ops) {
 
+        // Prevent memory leaks by removing dead listeners
+        this._server.events.removeListener('log', this._logHandler);
+        this._server.events.removeListener('response', this._responseHandler);
+        // These cannot be removed because Podium does not support unsubscribing from criteria, only event names.
+        // this._server.events.removeListener({ name: 'request', channels: ['error'] }, this._errorHandler);
+        // this._server.events.removeListener({ name: 'request', channels: ['app'] }, this._requestLogHandler);
```

Comment:
Just realized it may be possible that these produce duplicate events if registered more than once. Will need to double check on that.

Reply:
Turns out it would create duplicates. I've fixed and added a test to verify this doesn't happen.